public/actions.js

```diff
@@ -436,6 +436,34 @@ export const AddGoal = (state) => {
     }),
   ];
 };
+export const AddMultipleGoals = (state, goals) => {
+  const allGoals = state.goals.concat(
+    goals
+      .split(";")
+      .map(text => text.trim())
+      .filter(text => text.length > 0)
+      .map(text => {
+        return {
+          id: Math.random().toString(36).slice(2),
+          text: text,
+          completed: false,
+        };
+      })
+  );
+
+  return [
+    {
+      ...state,
+      goals: allGoals,
+      goal: '',
+    },
+    effects.UpdateGoals({
+      websocket: state.websocket,
+      allGoals,
```

Comment:
Unfortunately with plain-old js, I think things got lost in translation, this should be `goals: allGoals`.

Reply:
Good catch!